packages/transformers/posthtml/src/PostHTMLTransformer.js

```diff
@@ -9,23 +9,71 @@ import nullthrows from 'nullthrows';
 import semver from 'semver';
 import loadPlugins from './loadPlugins';
 
-export default new Transformer({
-  async getConfig({asset, options}) {
-    let config = await asset.getConfig(
-      ['.posthtmlrc', '.posthtmlrc.js', 'posthtml.config.js'],
-      {
-        packageKey: 'posthtml',
-      },
+const canSerializeConfig = config => {
+  if (!config || !config.plugins) {
+    return true;
+  }
+  if (Array.isArray(config.plugins)) {
+    return config.plugins.every(plugin => typeof plugin === 'string');
+  } else if (typeof config.plugins === 'object') {
+    return Object.keys(config.plugins).every(
+      plugin => typeof plugin === 'string',
     );
+  }
+  return true;
+};
 
-    config = config || {};
+export default new Transformer({
+  async loadConfig({config, options}) {
+    let loaded = await config.getConfig([
+      '.posthtmlrc',
+      '.posthtmlrc.js',
+      'posthtml.config.js',
+    ]);
+
+    loaded = loaded || {};
+    loaded.skipParse = true;
+
+    if (!canSerializeConfig(loaded)) {
+      config.setResult({fullyLoaded: loaded});
+      config.shouldInvalidateOnStartup();
+      config.setResultHash(JSON.stringify(Date.now()));
+    } else {
+      let fullyLoaded = {
+        ...loaded,
+        plugins: await loadPlugins(loaded.plugins, config.searchPath, options),
```

Comment:
@padmaia, I am sure we want to run this during `loadConfig`. Since this will make the config unserializable. 

I think the [`postDeserializeConfig`](https://github.com/parcel-bundler/parcel/pull/3565/files#diff-e170e4a4898565038389b704114bb0acR74), is covering this.